### DISCUSSION TITLE
Best practice or template to interface with the inventory?

### DISCUSSION THREAD

**Discussion by BigBroken** [peter-kish/gloot#104] [Q&A] [answered]
I'm looking for a basic example on how to interface with inventory. Like for example using a potion by clicking on it.
I would expect there to be a signal that fires that other scripts could subscribe to? it's certainly possible to make work by updating the existing codebase, but wondering if there is a cleaner way to interface with this addon?
Thanks

**Comment by peter-kish** [comment, marked as answer]
Hi!
Connecting to the `inventory_item_acitivated` signal should do the trick. It's defined for the `CtrlInventory`, `CtrlInventoryGrid` and `CtrlInventoryStacked` classes and gets emitted when an item is double-clicked in the inventory. 

I should probably update the examples to demonstrate this, as it is a pretty common mechanic.

Another way of doing this could be to get all the selected items with `get_selected_inventory_items() ` and do something with them when clicking a button or pressing a key.
